### ISSUE TITLE
Nx-Cloud - Axios does not respect proxy configuration (env) or proxy configuration issue

### ISSUE THREAD

**Issue by rryter** [nrwl/nx#16173]
### Current Behavior

Axios does not respect the HTTP(S)_PROXY and NO_PROXY environment variables. 

When running `npx nx g @nrwl/nx-cloud:init`, it fails with:  `>  NX   socket hang up`

### Expected Behavior

It respects the environment variables and the communication to cloud.nx.app is successful.

### Steps to Reproduce

1. Setup a proxy HTTP and HTTPS
2. Try to run `npx nx g @nrwl/nx-cloud:init`
3. It fails with:  "**>  NX   socket hang up**"

Maybe it's also a proxy configuration issue? Otherwise this would have been reported already I guess.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 18.15.0
   OS   : linux x64
   npm  : 9.5.0
   
   nx                      : 15.8.5
   @nrwl/js                : 15.8.5
   @nrwl/jest              : 15.8.5
   @nrwl/linter            : 15.8.5
   @nrwl/workspace         : 15.8.5
   @nrwl/angular           : 15.8.5
   @nrwl/cli               : 15.8.5
   @nrwl/cypress           : 15.8.5
   @nrwl/devkit            : 15.8.5
   @nrwl/eslint-plugin-nx  : 15.8.5
   @nrwl/storybook         : 15.8.5
   @nrwl/tao               : 15.8.5
   @nrwl/webpack           : 15.8.5
   @nrwl/nx-cloud          : 15.3.5
   typescript              : 4.9.5
   ---------------------------------------
   Community plugins:
   @ng-bootstrap/ng-bootstrap : 14.0.1
   @compodoc/compodoc         : 1.1.19
   @storybook/angular         : 7.0.0-rc.9
   ng-mocks                   : 14.7.3
```

### Failure Logs

```shell
❯ NODE_DEBUG=net,http,tls npx nx g @nrwl/nx-cloud:init --verbose
NET 7286: createConnection [
  { path: '/tmp/7c35230dce01176964b7/d.sock' },
  [Function (anonymous)],
  [Symbol(normalizedArgs)]: true
]
NET 7286: pipe true /tmp/7c35230dce01176964b7/d.sock
NET 7286: afterConnect
NET 7286: destroy
NET 7286: close
NET 7286: close handle
NET 7286: createConnection [
  { path: '/tmp/7c35230dce01176964b7/d.sock' },
  null,
  [Symbol(normalizedArgs)]: true
]
NET 7286: pipe true /tmp/7c35230dce01176964b7/d.sock
NET 7286: _read
NET 7286: _read wait for connection
NET 7286: emit close
NET 7286: afterConnect
NET 7286: _read
NET 7286: Socket._handle.readStart
NET 7286: _read
...
NET 7286: _read

>  NX  Generating @nrwl/nx-cloud:init

HTTP 7286: call onSocket 0 0
HTTP 7286: createConnection www-proxy.{$company}.ch:8080: [Object: null prototype] {
  maxRedirects: 21,
  maxBodyLength: 10485760,
  protocol: 'http:',
  path: null,
  method: 'POST',
  headers: {
    Accept: 'application/json, text/plain, */*',
    'Content-Type': 'application/json',
    'User-Agent': 'axios/0.21.4',
    'Content-Length': 59,
    host: 'cloud.nx.app'
  },
  agent: undefined,
  agents: { http: undefined, https: undefined },
  auth: undefined,
  hostname: 'www-proxy.{$company}.ch',
  port: '8080',
  beforeRedirect: [Function: beforeRedirect],
  nativeProtocols: {
    'http:': {
      _connectionListener: [Function: connectionListener],
      METHODS: [Array],
      STATUS_CODES: [Object],
      Agent: [Function],
      ClientRequest: [Function: ClientRequest],
      IncomingMessage: [Function: IncomingMessage],
      OutgoingMessage: [Function: OutgoingMessage],
      Server: [Function: Server],
      ServerResponse: [Function: ServerResponse],
      createServer: [Function: createServer],
      validateHeaderName: [Function: __node_internal_],
      validateHeaderValue: [Function: __node_internal_],
      get: [Function: get],
      request: [Function: request],
      setMaxIdleHTTPParsers: [Function: setMaxIdleHTTPParsers],
      maxHeaderSize: [Getter],
      globalAgent: [Getter/Setter]
    },
    'https:': {
      Agent: [Function: Agent],
      globalAgent: [Agent],
      Server: [Function: Server],
      createServer: [Function: createServer],
      get: [Function: get],
      request: [Function: request]
    }
  },
  pathname: 'https://cloud.nx.app/nx-cloud/create-org-and-workspace',
  host: 'www-proxy.{$company}.ch',
  noDelay: true,
  servername: 'cloud.nx.app',
  _agentKey: 'www-proxy.{$company}.ch:8080:'
}
NET 7286: createConnection [
  [Object: null prototype] {
    maxRedirects: 21,
    maxBodyLength: 10485760,
    protocol: 'http:',
    path: null,
    method: 'POST',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      'User-Agent': 'axios/0.21.4',
      'Content-Length': 59,
      host: 'cloud.nx.app'
    },
    agent: undefined,
    agents: { http: undefined, https: undefined },
    auth: undefined,
    hostname: 'www-proxy.{$company}.ch',
    port: '8080',
    beforeRedirect: [Function: beforeRedirect],
    nativeProtocols: { 'http:': [Object], 'https:': [Object] },
    pathname: 'https://cloud.nx.app/nx-cloud/create-org-and-workspace',
    host: 'www-proxy.{$company}.ch',
    noDelay: true,
    servername: 'cloud.nx.app',
    _agentKey: 'www-proxy.{$company}.ch:8080:',
    encoding: null
  },
  [Function (anonymous)],
  [Symbol(normalizedArgs)]: true
]
NET 7286: pipe false null
NET 7286: connect: find host www-proxy.{$company}.ch
NET 7286: connect: dns options { family: undefined, hints: 32 }
HTTP 7286: sockets www-proxy.{$company}.ch:8080: 1 1
HTTP 7286: write ret = true
(node:7286) Warning: Setting the NODE_DEBUG environment variable to 'http' can expose sensitive data (such as passwords, tokens and authentication headers) in the resulting log.
(Use `node --trace-warnings ...` to show where the warning was created)
NET 7286: _read
NET 7286: _read wait for connection
NET 7286: afterConnect
HTTP 7286: outgoing message end.
NET 7286: _read
NET 7286: Socket._handle.readStart
NET 7286: destroy
NET 7286: close
NET 7286: close handle

 >  NX   socket hang up
```

### Additional Information

Updating 

```js
function createNxCloudWorkspace(workspaceName, installationSource) {
    return __awaiter(this, void 0, void 0, function* () {
        const apiUrl = removeTrailingSlash(process.env.NX_CLOUD_API || process.env.NRWL_API || `https://cloud.nx.app`);
        const response = yield require('axios').post(`${apiUrl}/nx-cloud/create-org-and-workspace`, {
            workspaceName,
            installationSource,
        });
        if (response.data.message) {
            throw new Error(response.data.message);
        }
        return response.data;
    });
}
```

to 
```js
const httpsProxyAgent = require('https-proxy-agent');
const httpsAgent = new httpsProxyAgent(`http://www-proxy.{$company}.ch:8080`);

function createNxCloudWorkspace(workspaceName, installationSource) {
    return __awaiter(this, void 0, void 0, function* () {
        const apiUrl = removeTrailingSlash(process.env.NX_CLOUD_API || process.env.NRWL_API || `https://cloud.nx.app`);
        const response = yield require('axios').post(`${apiUrl}/nx-cloud/create-org-and-workspace`, {
            workspaceName,
            installationSource,
        },{
            httpsAgent: httpsAgent,
            proxy: false,
          });
        if (response.data.message) {
            throw new Error(response.data.message);
        }
        return response.data;
    });
}
```

That "solves" the problem. The problem then pops up at every subsequent axios call to the cloud though.

Edit: 

This is the output for `npm list axios`:
```
├─┬ @nrwl/nx-cloud@15.3.5
│ └── axios@0.21.4
├─┬ @storybook/test-runner@0.9.4
│ └─┬ jest-playwright-preset@2.0.0
│   └─┬ jest-process-manager@0.3.1
│     └─┬ wait-on@5.3.0
│       └── axios@0.21.4
└─┬ nx@15.8.5
  └── axios@1.3.4
```` 

Note the difference in terms of versions between `nx-cloud` and `nx`

**Comment by StalkAltan** [comment]
@rryter Thanks for reporting this -- we coincidentally released a beta on Friday providing support for users that want to pass their own proxy configurations (via the `hpagent` package). We are going to test that with some of our enterprise clients before getting the docs and full release.

We will take a look into the cloud init generator not respecting env variables, or provide some way to provide a proxy to that initial call.

Thanks for your patience here, I'd expect something in the next week or so. We'll keep this issue updated.

**Comment by rryter** [comment]
Thank you very much for getting back to me so quickly. If you need help verifying that it works as intended, I'm happy to do some tests since we have the usual enterprise setup here. I'm looking forward to hearing from you.

**Comment by StalkAltan** [comment]
@rryter This has been released in version 16.0.0 of the Nx Cloud package, which should be backwards compatible with most versions of Nx back to around 13. 

Docs are available here: https://github.com/nrwl/nx-cloud-helm/blob/main/PROXY-GUIDE.md#nxcloud-runner-proxy-issues
